### PR TITLE
Jormun: Add data for shared sections and adapt tests 

### PR DESCRIPTION
**Pull request by mbounabi**

### CODE REVIEW

**Review by benoit-bst — APPROVED** [review]

File: source/jormungandr/tests/disruptions_tests.py [hunk 1]
```diff
@@ -189,48 +190,95 @@ def test_disruption_with_stop_areas(self):
         assert len(disruptions) == 1
         is_valid_disruption(disruptions[0])
 
-    def test_disruption_with_departures(self):
+    def is_in_disprution_time(self, stop_time, disruption_time):
```
Comment (xlqian) [inline source/jormungandr/tests/disruptions_tests.py:193]: so your function never returns false ? LOL IMHO, you can  do it in a better way :)

File: source/jormungandr/tests/disruptions_tests.py [hunk 2]
```diff
@@ -189,48 +190,95 @@ def test_disruption_with_stop_areas(self):
...
-        is_valid_disruption(disruptions['too_bad_all_lines'][0].disruption)
+        assert len(disruptions) >= 1
+        for d in disruptions:
+            is_valid_disruption(d)
 
-        departure = departures[1]
-        disruptions = get_all_element_disruptions(departure, response)
-        assert not disruptions
+        # Every lines impacted by the disruption
+        disruption_lines = jmespath.search('impacted_objects[].pt_object.line.name', disruptions[0])
```
Comment (xlqian) [inline source/jormungandr/tests/disruptions_tests.py:219]: :+1:

File: source/jormungandr/tests/heat_maps_tests.py [hunk 3]
```diff
@@ -50,7 +50,35 @@ def get_duration(coord, response):
 class TestHeatMap(AbstractTestFixture):
     """
     Test the structure of the heat_maps response
-    Only the format of the heat_map is check not the value.
+    Only the format of the heat_map is check not the value. The value returned is an approximation, as explained below.
```
Comment (xlqian) [inline source/jormungandr/tests/heat_maps_tests.py:53]: :+1:

File: source/jormungandr/tests/kirin_realtime_tests.py [hunk 4]
```diff
@@ -127,8 +129,8 @@ def _check_train_cancel_disruption(dis):
         assert vjs[0]['id'] == 'vjA'
 
         new_response = self.query_region(journey_basic_query + "&data_freshness=realtime")
-        assert get_arrivals(new_response) == ['20120614T080436', '20120614T180222']
-        assert get_used_vj(new_response) == [[], ['vjB']]
+        assert set(get_arrivals(new_response)) == set(['20120614T080436', '20120614T080223'])
+        assert get_used_vj(new_response) == [['vjM'], []]
```
Comment (xlqian) [inline source/jormungandr/tests/kirin_realtime_tests.py:133]: if you want the 'unordered-ness ', you can use tuple which is hashable in python

File: source/jormungandr/tests/disruptions_tests.py (hunk 1 above)
Comment (mbounabi) [reply source/jormungandr/tests/disruptions_tests.py:193]: well, it returns None by default. I can add the else: return False if needed but the result will be the same.

Comment (xlqian) [reply source/jormungandr/tests/disruptions_tests.py:193]: can you just do

```python
return disruption_time['begin'] < stop_time['arrival_date_time'] < disruption_time['end'] or \
 disruption_time['begin'] < stop_time['departure_date_time'] < disruption_time['end']
```

File: source/jormungandr/tests/kirin_realtime_tests.py (hunk 4 above)
Comment (mbounabi) [reply source/jormungandr/tests/kirin_realtime_tests.py:133]: But tuples are ordered, so it won't work.

Comment (xlqian) [reply source/jormungandr/tests/kirin_realtime_tests.py:133]: ok

Comment (xlqian) [reply source/jormungandr/tests/kirin_realtime_tests.py:133]: just for fun, you can take a look at frozenset

**Review by xlqian — APPROVED** [review]

File: source/routing/tests/routing_api_test_data.h [hunk 5]
```diff
@@ -420,6 +420,13 @@ struct routing_api_data {
             b.data->pt_data->headsign_handler.affect_headsign_to_stop_time(
                                 b.data->pt_data->vehicle_journeys.at(0)->stop_time_list.at(0), "A00");
 
+            b.vj("M", "111111", "", false, "vjM")("stop_point:stopB", "08:01:01"_t)("stop_point:stopA", "08:01:03"_t)
+                                        .st_shape({B, I, A});
+			b.lines["M"]->code = "1M";
+			b.lines["M"]->color = "3ACCDC";
+			b.lines["M"]->text_color = "FFFFFF";
```
Comment (nonifier) [inline source/routing/tests/routing_api_test_data.h:427]: Why do you need to set `code/color/text_color` here ?

Comment (mbounabi) [reply source/routing/tests/routing_api_test_data.h:427]: Because it looks cool in the Playground! :)

**Review by nonifier — APPROVED** [review]

File: source/routing/tests/routing_api_test_data.h [hunk 6]
```diff
@@ -420,6 +420,13 @@ struct routing_api_data {
...
+			//b.data->pt_data->headsign_handler.affect_headsign_to_stop_time(b.data->pt_data->vehicle_journeys.at(0)->stop_time_list.at(0), "A00");
```
Comment (pbougue) [inline source/routing/tests/routing_api_test_data.h:428]: Why this commented line ?

File: source/jormungandr/tests/kirin_realtime_tests.py [hunk 7]
```diff
@@ -300,14 +306,14 @@ def _check_train_delay_disruption(dis):
...
-        assert len(pt_response['vehicle_journeys']) == 7
+        assert len(pt_response['vehicle_journeys']) == (initial_nb_vehicle_journeys + 1)
 
         pt_response = self.query_region('vehicle_journeys/vjA?_current_datetime=20120614T1337')
         assert len(pt_response['disruptions']) == 1
         _check_train_delay_disruption(pt_response['disruptions'][0])
 
         # so the first real-time vj created for the first disruption should be deactivated
-        new_response = self.query_region(journey_basic_query + "&data_freshness=realtime")
+        new_response = self.query_region(journey_basic_query + "&data_freshness=realtime&forbidden_uris[]=M&")
```
Comment (pbougue) [inline source/jormungandr/tests/kirin_realtime_tests.py:316]: The same comment on why M is forbidden could be nice.

File: source/jormungandr/tests/disruptions_tests.py [hunk 8]
```diff
@@ -189,48 +190,94 @@ def test_disruption_with_stop_areas(self):
...
+        # Every lines departing from the stop point
+        departures_lines = jmespath.search('[].route.line.name', departures)
+
+        # Line not impacted shouldn't return a disruption
+        not_impacted_lines = set(departures_lines)-set(disruption_lines)
```
Comment (pbougue) [inline source/jormungandr/tests/disruptions_tests.py:224]: minor: spacing

**Review by pbougue — COMMENTED** [review]
Just formal: for an upcoming PR

File: source/routing/tests/routing_api_test_data.h (hunk 6 above)
Comment (mbounabi) [reply source/routing/tests/routing_api_test_data.h:428]: a forgotten line in the immensity of the code...
Now removed

File: source/jormungandr/tests/kirin_realtime_tests.py (hunk 7 above)
Comment (mbounabi) [reply source/jormungandr/tests/kirin_realtime_tests.py:316]: Now added

File: source/jormungandr/tests/disruptions_tests.py (hunk 8 above)
Comment (mbounabi) [reply source/jormungandr/tests/disruptions_tests.py:224]: spaced it out!

**Pull request merged**